kod/object/passive/itematt/weapatt/wapurge.kod

```diff
@@ -71,28 +71,32 @@ messages:
    %%% Effect Functions
       
    ModifyDamage(damage = 0, target = $, wielder = $, lData= $)
-   "Wielder has a 1% chance of blinding target with any hit."
+   "Wielder has a percent chance of purging target with any hit."
    {
-      local oSpell;
+      local oSpell, oRoom;
 
       if random(1,100) < piEffect_percent
       {
-         oSpell = send(SYS,@FindSpellByNum,#num=SID_PURGE);
+         oSpell = Send(SYS,@FindSpellByNum,#num=SID_PURGE);
          
          %% Gotta have something to try to remove....
          if Send(target,@IsEnchanted)
          {
-            % Try to purge off random(25,75)% of the enchantments.
-            if send(oSpell,@DoPurge,#who=target,#iChance=random(25,75))
+            oRoom = Send(wielder,@GetOwner);
+
+            % Try to purge off random number of the enchantments
+            % Only tell our victim if they lost enchantments
+            if Send(oSpell,@DoPurge,#who=target,#iChance=random(25,75))
             {
-               % Only tell our victim if they lost enchantments.
-               send(target,@MsgSendUser,#message_rsc=purger_worked_target,
-                    #parm1=send(wielder,@GetCapDef),#parm2=send(wielder,@GetName));
+               Send(oSpell,@PlaySpellSound,#room_obj=oRoom,#what=wielder);
```

Comment:
Just checking that this isn't going to stack with existing sounds (hit, death, whatever) and play two at once.  ModifyDamage doesn't seem like the safest place to do this.

Reply:
I was thinking about refactoring to use `CastSpell` like some of the other spell driven weapon attributes. This would tap into the existing audio cues but would also have it fall into quite a bit of other logic as well. I wonder if this is an issue for these other weapon attributes?